_posts/2016/2016-05-31-gradle-jenkins-plugin.md

```diff
@@ -0,0 +1,112 @@
+
+Another advantage of this gradle plugin is that is simple to deploy. For instance I didn't want to break our Team Jenkins installation, so I run a instance of Jenkins in a docker container and deploy my playground scripts to it.
+
+## How to setup the project
+In your main Project you should use the [gradle-jenkins-plugin](https://github.com/ghale/gradle-jenkins-plugin/wiki).
+
+I created a sample snippet for writing this blog post:
+{% highlight groovy %}
+buildscript {
+  repositories {
+    maven { url "https://plugins.gradle.org/m2/" }
+    maven { url 'http://repo.jenkins-ci.org/releases/' }
+    mavenLocal()
+  }
+  dependencies {
+    classpath "com.terrafolio:gradle-jenkins-plugin:1.3.2"
+  }
+}
+
+apply plugin: 'com.terrafolio.jenkins'
+
+jenkins {
+
+    servers {
+        localhost_jenkins {
+            url 'http://localhost:8080/' //change for your jenkins name
+            secure true
+            username 'admin' //change for your user name
+            password 'admin' //change for your password
+        }
+    }
+
+    defaultServer servers.localhost_jenkins
+
+    dsl {
+        job("SampleJob") {
+            displayName 'My sample job'
+            description ''
+            logRotator {
+                numToKeep(10)
+            }
+        }
+
+    } // end dsl
+}
+{% endhighlight %}
+
+## How to use
+
+I was using on my local Jenkins. After doing the proper changes on the dsl (jenkins.gradle), I just need to trigger the task updateJenkinsItems.
+
+To use it, you should create a gradle project and copy the snippet.
+On the project folder you could use:
+
+-----------------------
+{% highlight bash %}
+./gradlew tasks
+{% endhighlight %}
+
+This will provide you the list of available tasks:
+{% image blog/blog-gradlew_tasks.jpg %}
+
+-----------------------
+{% highlight bash %}
+./gradlew validateJenkinsItems
+{% endhighlight %}
+This will validate local changes with your Jenkins
+
+-----------------------
+{% highlight bash %}
+./gradlew updateJenkinsItems
+{% endhighlight %}
+This will update your Jenkins installation with the dsl defined at your jenkins.gradle
+{% image blog/blog-gradlew_updateJenkinsItems.jpg %}
+
+
+## Conclusions
+This plugin is really powerful and there are many cases where you could use to empower your jobs/automation environment. I hope the blog post ilustrate the applications of this plugin.
+If you want to know more about the other jobs, have a look at complete documentation on the [project wiki](https://github.com/ghale/gradle-jenkins-plugin/wiki).
+
+#### Advantages
+- We have our Jenkins jobs under a source control system (git)
+- We can recreate our Jenkins instance pretty fast and test locally without affecting the rest of the team
+
+#### Disadvantages
+- If you are not familiar with Gradle, it might take some time for you to get confident with the configuration
```

Comment:
If you are not familiar with Gradle, it might take some time for you to get confident with the configuration
>  If you are not familiar with Gradle, it might take some time for you to become acquainted with the configuration.